src/TriggerAssigner.cc

```diff
@@ -0,0 +1,136 @@
+  mCheckPeriod(period, false);
+  mCheckChannel(channel);
+  return mLep1Thresh[period][channel];
+}
+
+float TriggerAssigner::getLepton2PtThresh(std::string period, std::string channel) {
+  mCheckPeriod(period, true);
+  mCheckChannel(channel);
+  return mLep2Thresh[period][channel];
+}
+
+// Legacy triggers take precedence over Single-Tau and MET triggers
+TriggerAssigner::uMap<bool>
+TriggerAssigner::assignTriggerRegions(int ptype, std::string period,
+									  float pt1, float pt2, float eta1, float eta2,
+									  float eleEtaMax, float muEtaMax, float tau_thresh)
+{
+  bool Legacy_region    = false;
+  bool SingleTau_region = false;
+  bool MET_region       = false;
+  
+  if (ptype == 0) { //mutau
+	bool SingleLepton_validity = false;
+	bool CrossLepton_validity  = false;
+	bool SingleTau_validity    = false;
+	
+	if(period=="2018") {
+	  SingleLepton_validity = pt1 >= mLep1Thresh[period]["mutau"] and eta1 < muEtaMax;
+	  CrossLepton_validity	= (pt1 >= 22. and eta1 < mEtaThresh and
+							   pt2 >= mLep2Thresh[period]["mutau"] and eta2 < mEtaThresh);
+	  SingleTau_validity	= pt2 >= tau_thresh and eta2 < mEtaThresh;
+	}
+	else if(period=="2017") {
+	  SingleLepton_validity = pt1 >= mLep1Thresh[period]["mutau"] and eta1 < muEtaMax;
+	  CrossLepton_validity	= (pt1 >= 22. and eta1 < mEtaThresh and
+							   pt2 >= mLep2Thresh[period]["mutau"] and eta2 < mEtaThresh);
+	  SingleTau_validity	= pt2 >= tau_thresh and eta2 < mEtaThresh;
+	}
+	else if (period=="2016preVFP" or period=="2016postVFP") {
+	  SingleLepton_validity = pt1 >= mLep1Thresh[period]["mutau"] and eta1 < muEtaMax;
+	  CrossLepton_validity	= (pt1 >= 21. and eta1 < mEtaThresh and
+							   pt2 >= mLep2Thresh[period]["mutau"] and eta2 < mEtaThresh);
+	  SingleTau_validity	= pt2 >= tau_thresh and eta2 < mEtaThresh;
+	}
+
+	Legacy_region = SingleLepton_validity or CrossLepton_validity;
+	SingleTau_region = SingleTau_validity and !Legacy_region;
+  }
+
+  else if (ptype == 1) { //etau
+	if(period=="2018" or period=="2017") {
+	  bool SingleLepton_validity = pt1 >= mLep1Thresh[period]["etau"] and eta1 < eleEtaMax;
+	  bool CrossLepton_validity  = (pt1 >= 25. and eta1 < mEtaThresh and
+									pt2 >= mLep2Thresh[period]["etau"] and eta2 < mEtaThresh);
+	  bool SingleTau_validity    = pt2 >= tau_thresh and eta2 < mEtaThresh;
+
+	  Legacy_region = SingleLepton_validity or CrossLepton_validity;
+	  SingleTau_region = SingleTau_validity and !Legacy_region;
+	}
+	else if (period=="2016preVFP" or period=="2016postVFP") {
+	  Legacy_region    = pt1 >= mLep1Thresh[period]["etau"] and eta1 < mEtaThresh;
+	  SingleTau_region = pt2 >= tau_thresh and !Legacy_region;
+	}
+  }
+
+  else if (ptype == 2) { //tautau
+	Legacy_region = pt1 >= 40. and pt2 >= 40. and eta1 < mEtaThresh and eta2 < mEtaThresh;
+	SingleTau_region = (((pt1 >= tau_thresh and eta1 < mEtaThresh) or (pt2 >= tau_thresh and eta2 < mEtaThresh))
+						and !Legacy_region);
+  }
+  
+  if (ptype == 3) { //mumu
```

Comment:
not that it really matter,  but for consistency could you change that to an `else if` statement?